packages/enzyme-adapter-react-17/src/ReactSeventeenAdapter.js

```diff
@@ -0,0 +1,895 @@
+        // It's conceivable that testers only want to assert on the component that's rendered i.e. only fallback or only component but never both.
+        : [childrenToTree(node.child.child), childrenToTree(node.child.sibling)];
+      return {
+        nodeType: 'function',
+        type: Suspense,
+        props: { ...node.memoizedProps },
+        key: ensureKeyOrUndefined(node.key),
+        ref: node.ref,
+        instance: null,
+        rendered,
+      };
+    }
+    case FiberTags.Lazy:
+      return childrenToTree(node.child);
+    case FiberTags.OffscreenComponent: {
+      throw new Error('Enzyme Internal Error. Encountered a Offscreen Fiber');
+    }
+    default:
+      throw new Error(`Enzyme Internal Error: unknown node with tag ${node.tag}`);
+  }
+}
+
+function childrenToTree(node) {
+  if (!node) {
+    return null;
+  }
+  const children = nodeAndSiblingsArray(node);
+  if (children.length === 0) {
+    return null;
+  }
+  if (children.length === 1) {
+    return toTree(children[0]);
+  }
+  return flatten(children.map(toTree));
+}
+
+function nodeToHostNode(_node) {
+  // NOTE(lmr): node could be a function component
+  // which wont have an instance prop, but we can get the
+  // host node associated with its return value at that point.
+  // Although this breaks down if the return value is an array,
+  // as is possible with React 16.
+  let node = _node;
+  while (node && !Array.isArray(node) && node.instance === null) {
+    node = node.rendered;
+  }
+  // if the SFC returned null effectively, there is no host node.
+  if (!node) {
+    return null;
+  }
+
+  const mapper = (item) => {
+    if (item && item.instance) return ReactDOM.findDOMNode(item.instance);
+    return null;
+  };
+  if (Array.isArray(node)) {
+    return node.map(mapper);
+  }
+  if (Array.isArray(node.rendered) && node.nodeType === 'class') {
+    return node.rendered.map(mapper);
+  }
+  return mapper(node);
+}
+
+function replaceLazyWithFallback(node, fallback) {
+  if (!node) {
+    return null;
+  }
+  if (Array.isArray(node)) {
+    return node.map((el) => replaceLazyWithFallback(el, fallback));
+  }
+  if (isLazy(node.type)) {
+    return fallback;
+  }
+  return {
+    ...node,
+    props: {
+      ...node.props,
+      children: replaceLazyWithFallback(node.props.children, fallback),
+    },
+  };
+}
+
+const eventOptions = {
+  animation: true,
+  pointerEvents: true,
+  auxClick: true,
+};
+
+function wrapAct(fn) {
+  let returnVal;
+  TestUtils.act(() => { returnVal = fn(); });
+  return returnVal;
+}
+
+function getProviderDefaultValue(Provider) {
+  if ('_currentValue' in Provider._context) {
+    return Provider._context._currentValue;
+  }
+  throw new Error('Enzyme Internal Error: can’t figure out how to get Provider’s default value');
+}
+
+function makeFakeElement(type) {
+  return { $$typeof: Element, type };
+}
+
+function isStateful(Component) {
+  return Component.prototype && Component.prototype.isReactComponent;
```

Comment:
@ljharb You're probably referring to this change compared to 16 regarding `create-react-class` changes?

Older versions of `create-react-class` might not pass this duck-typing check but then they would also not pass the same check in React 17 itself: 
- https://github.com/facebook/react/blob/v17.0.2/packages/react-reconciler/src/ReactFiber.new.js#L220
- https://github.com/facebook/react/blob/v17.0.2/packages/react-reconciler/src/ReactFiber.new.js#L457-L458

Reply:
Thanks for confirming :-)